inference-engine/src/mkldnn_plugin/nodes/mkldnn_mvn_node.cpp

```diff
@@ -672,135 +812,28 @@ void MKLDNNMVNNode::setPostOps(mkldnn::primitive_attr &attr, bool initWeights) {
-                mvn_blk<int8_t, float>(src_data, dst_data, getParentEdgeAt(0)->getDesc().getDims());
-            } else if (input_prec == Precision::FP32) {
-                auto src_data = reinterpret_cast<float *>(srcMemPtr->GetData());
-                mvn_blk<float, float>(src_data, dst_data, getParentEdgeAt(0)->getDesc().getDims());
-            } else if (input_prec == Precision::BF16) {
-                auto src_data = reinterpret_cast<const bfloat16_t*>(srcMemPtr->GetData());
-                mvn_blk<bfloat16_t, float>(src_data, dst_data, getParentEdgeAt(0)->getDesc().getDims());
-            } else {
-                THROW_IE_EXCEPTION << "Unsupported input precision: " << input_prec.name();
-            }
-        } else if (output_prec == Precision::BF16) {
-            auto dst_data = reinterpret_cast<bfloat16_t*>(dstMemPtr->GetData());
-            if (input_prec == Precision::U8) {
-                auto src_data = reinterpret_cast<const uint8_t *>(srcMemPtr->GetData());
-                mvn_blk<uint8_t, bfloat16_t>(src_data, dst_data, getParentEdgeAt(0)->getDesc().getDims());
-            } else if (input_prec == Precision::I8) {
-                auto src_data = reinterpret_cast<const int8_t *>(srcMemPtr->GetData());
-                mvn_blk<int8_t, bfloat16_t>(src_data, dst_data, getParentEdgeAt(0)->getDesc().getDims());
-            } else if (input_prec == Precision::FP32) {
-                auto src_data = reinterpret_cast<float *>(srcMemPtr->GetData());
-                mvn_blk<float, bfloat16_t>(src_data, dst_data, getParentEdgeAt(0)->getDesc().getDims());
-            } else if (input_prec == Precision::BF16) {
-                auto src_data = reinterpret_cast<const bfloat16_t*>(srcMemPtr->GetData());
-                mvn_blk<bfloat16_t, bfloat16_t>(src_data, dst_data, getParentEdgeAt(0)->getDesc().getDims());
-            } else {
-                THROW_IE_EXCEPTION << "Unsupported input precision: " << input_prec.name();
-            }
-        } else {
-            THROW_IE_EXCEPTION << "Unsupported output precision: " << output_prec.name();
-        }
-    }
-}
-
-std::tuple<size_t, size_t, size_t, size_t, size_t> MKLDNNMVNNode::get5dShapes(const SizeVector& dims) {
-    std::tuple<size_t, size_t, size_t, size_t, size_t> shapes;
-    switch (dims.size()) {
-        case 1 : { shapes = std::make_tuple(1, dims[0], 1, 1, 1); break; }
-        case 2 : { shapes = std::make_tuple(dims[0], dims[1], 1, 1, 1); break; }
-        case 3 : { shapes = std::make_tuple(dims[0], dims[1], 1, dims[2], 1); break; }
-        case 4 : { shapes = std::make_tuple(dims[0], dims[1], 1, dims[2], dims[3]); break; }
-        case 5 : { shapes = std::make_tuple(dims[0], dims[1], dims[2], dims[3], dims[4]); break; }
-        default : { THROW_IE_EXCEPTION << "MVN layer with name '" << getCnnLayer()->name << "' doesn't support planar layout with rank: " << dims.size(); }
+        mvn_blk(src_data, dst_data, dim);
```

Comment:
I don't wee there `mvn_ref` is used. Doesn't look like it is expected.

Reply:
fixed[ here](https://github.com/openvinotoolkit/openvino/pull/3567/files#diff-3593bcddbeb921a60602fad8865d89c2b8f74901bc4a28114c7bb976f625f8f7R866) and rebased on mvn-6. Thank you!